src/org/zaproxy/zap/extension/script/ScriptWrapper.java

```diff
@@ -292,4 +301,55 @@ void setPreviouslyEnabled(boolean enabled) {
 	boolean isPreviouslyEnabled() {
 		return previouslyEnabled;
 	}
+
+	/**
+	 * Reloads the script. It should only be called if the script has already been loaded via {@link #loadScript(Charset)}
+	 * @throws IOException
+	 * @since TODO add version
+	 */
+	public void reloadScript() throws IOException {
+		this.loadScript(charset);
+	}
+
+	/**
+	 * Loads the script from the configured file.
+	 * @param charset
+	 * @throws IOException
+	 * @since TODO add version
+	 */
+	void loadScript(Charset charset) throws IOException {
+		this.charset = charset;
+		StringBuilder sb = new StringBuilder();
+		try (BufferedReader br = Files.newBufferedReader(file.toPath(), charset)) {
+			int len;
+			char[] buf = new char[1024];
+			while ((len = br.read(buf)) != -1) {
+				sb.append(buf, 0, len);
+			}
+		}
+		setContents(sb.toString());
+		setChanged(false);
+		this.lastModified = file.lastModified();
+	}
+	
+	/**
+	 * Saves the latest contents into the configured file
+	 * @throws IOException
+	 * @since TODO add version
+	 */
+	void saveScript() throws IOException {
+		try ( BufferedWriter bw = Files.newBufferedWriter(file.toPath(), charset)) {
```

Comment:
I liked the previous behaviour more :) We should save with the default always because we will try to read with that in the next ZAP run. We just need to update it for following (re)loads.

Reply:
Updated